src/preprocessor.py

```diff
@@ -565,11 +672,11 @@ def __init__(self, data_mgr, pod):
         # initialize xarray parser
         self.parser = self._XarrayParserClass(data_mgr, pod)
         # initialize PreprocessorFunctionBase objects
-        self.functions = [cls_(data_mgr, pod) for cls_ in self._functions]
+        self.functions = [cls_(data_mgr, pod) for cls_ in self.functions]
 
     @property
-    def _functions(self):
-        """Determine which preprocessor functions are applicable to the current
+    def functions(self):
+        """Determine which PreprocessorFunctions are applicable to the current
```

Comment:
Removing the `_` from the functions definition and reference in the list results in a Nonetype for the pod.preprocessor in the setup_pod routine, which causes the CI to fail. Reverting it should solve the issue.

Reply:
Thanks for diagnosing the source of the error so quickly, @wrongkindofdoctor! Reverted in [130e994](https://github.com/NOAA-GFDL/MDTF-diagnostics/pull/372/commits/130e994fcaf3a78a0ec4188e2a375e4c0f5cc5d9).

Because the attributes are private, their docstrings won't be included in the Sphinx autodoc docs (which I'm guessing is why I changed them in the first place), but that's a minor difference.